apps/nextjs/src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -1,18 +1,42 @@
-import type { NextRequest } from "next/server";
 import { cookies } from "next/headers";
-import { NextResponse } from "next/server";
+import { NextRequest, NextResponse } from "next/server";
 
-import { GET as DEFAULT_GET, POST } from "@acme/auth";
+import { GET as DEFAULT_GET, POST as DEFAULT_POST } from "@acme/auth";
+
+import { env } from "~/env";
 
 export const runtime = "edge";
 
 const EXPO_COOKIE_NAME = "__acme-expo-redirect-state";
 const AUTH_COOKIE_PATTERN = /authjs\.session-token=([^;]+)/;
 
+/**
+ * Correct request.url for local development so that Expo can work. Does nothing in production.
+ * @param req The request to modify
+ * @returns The modified request.
+ */
+function rewriteRequestUrl(req: NextRequest) {
+  if (env.VERCEL) {
```

Comment:
Should the case of an app not being deployed to Vercel be covered? I know most of these apps will be deployed there, but wouldn't be safer to check the `NODE_ENV`'s value?

Or if this is related to the `trustHost` configuration in [`packages/auth/src/config.ts#L35`](https://github.com/t3-oss/create-t3-turbo/pull/1054/files#diff-8643fbe71ccca0ed5101e1447adcd63902a0ea9daa4eaab4655c37ecb3875003R35), could it be reused here?

Reply:
It looks like I can just use `NODE_ENV === "production"` and it will handle vercel preview envs just fine, so I will move to  that check instead.